src/frontend/__mocks__/zustand.js

```diff
@@ -19,12 +21,23 @@ const create = (createState) => {
     }
     store.setState(initialState, true);
   });
+
+  stores.add(store);
+
   return store;
 };
 
 // Reset all stores after each test run
 afterEach(() => {
-  act(() => storeResetFns.forEach((resetFn) => resetFn()));
+  act(() => {
+    //  remove all listeners
+    //  we have to do this in case initial method of a store can throw
+    //  because when the state is reset, it will still trigger update on
+    //  listening components that may causes the throw to happend.
+    stores.forEach((store) => store.destroy());
```

Comment:
Trick to remove listener preventing the setState to trigger a rerender of components (else testing a component calling getDecodedJwt throws).